browser/brave_content_browser_client.cc

```diff
@@ -641,21 +637,21 @@ bool BraveContentBrowserClient::CanCreateWindow(
       container_type, target_url, referrer, frame_name, disposition, features,
       user_gesture, opener_suppressed, no_javascript_access);
 
-#if BUILDFLAG(ENABLE_REQUEST_OTR)
-  // If the user has requested going off-the-record in this tab, don't allow
+  // If the user has requested going off-the-record on this site, don't allow
   // opening new windows via script
   if (content::WebContents* web_contents =
           content::WebContents::FromRenderFrameHost(opener)) {
-    if (request_otr::
-            RequestOTRStorageTabHelper* request_otr_storage_tab_helper =
-                request_otr::RequestOTRStorageTabHelper::FromWebContents(
-                    web_contents)) {
-      if (request_otr_storage_tab_helper->has_requested_otr()) {
+    if (request_otr::RequestOTRService* request_otr_service =
+            request_otr::RequestOTRServiceFactory::GetForBrowserContext(
+                web_contents->GetBrowserContext())) {
+      // TODO(pilgrim) this doesn't work, url is about:blank. Are we already
```

Comment:
as discussed in DM you are correct, we get this for free by using a different storage partition